java/component/keyple-calypso/src/main/java/org/eclipse/keyple/calypso/transaction/CalypsoPo.java

```diff
@@ -453,6 +454,30 @@ public boolean isDfInvalidated() {
         return isDfInvalidated;
     }
 
+    /**
+     * Indicated whether the last session with this PO has been ratified or not.
+     * <p>
+     *
+     * @return true if the PO has been ratified.
+     * @throws IllegalStateException if these methods is call when no session has been opened
+     */
+    public boolean isDfRatified() {
+        if (isDfRatified != null) {
+            return isDfRatified;
+        }
+        throw new IllegalStateException(
+                "Unable to determine the ratification status. No session was opened.");
+    }
+
+    /**
+     * Set the ratification status
+     * 
+     * @param dfRatified true if the session was ratified
+     */
+    public void setDfRatified(boolean dfRatified) {
```

Comment:
Should we define the visibility as 'package' for the method CalypsoPo.setDfRatified(boolean) ?

Reply:
Absolutely!